tests/pipeline/test_pipeline.py

```diff
@@ -791,6 +791,19 @@ def test_only_nodes_unknown(self, pipeline_list_with_lists, target_node_names):
         with pytest.raises(ValueError, match=pattern):
             full.only_nodes(*target_node_names)
 
+    @pytest.mark.parametrize(
+        "dummy_namespaced_nodes", ["node1", "node2", "node3", "node4", "node5", "node6"]
+    )
+    def test_only_nodes_with_namespacing(
```

Comment:
I appreciate this test is written in quite a clever way, but it's also quite hard to understand what exactly is going on, as in it's not clear what the error message will actually contain. I'm also wondering if this should've caught the issue I found by manually testing and if not, it would be nice to have a test for the multiple node use cases I tried.

Reply:
Good point! I'll add some tests to check for multiple correct arguments, and multiple arguments where one is correct and one is not, to make sure all bases are covered.